### PR TITLE
Delete usages/tests of the builtin bazel java symbols

### PR DESCRIPTION
**Pull request by hvadehra**

### CODE REVIEW

**Review by tetromino — CHANGES_REQUESTED** [review]
The purpose of these tests is to exercise doc extraction from native/builtin symbols. If these symbols are now ordinary starlark symbols, the tests are not needed.

So let's simply remove java_basic_input test, and remove the use of JavaInfo from providers_for_attributes_test. (And then regenerate the golden files.)

Do you want to do this, or should I?

File: test/testdata/providers_for_attributes_test/input.bzl [hunk 1]
```diff
@@ -39,7 +39,7 @@ my_rule = rule(
             providers = [ProtoInfo, DefaultInfo, JavaInfo],
```
Comment (tetromino) [inline test/testdata/providers_for_attributes_test/input.bzl:39]: (Nit: also JavaInfo here.)

**Review by tetromino — CHANGES_REQUESTED** [review]

File: test/testdata/providers_for_attributes_test/input.bzl (hunk 1 above)
Comment (hvadehra) [reply test/testdata/providers_for_attributes_test/input.bzl]: Done

**Comment by tetromino** [comment]
@hvadehra - run `bazel run //test:providers_for_attributes_test_regenerate` to update the golden .md file.

**Comment by hvadehra** [comment]
The tests are already passing? I ran it anyway, did nothing.

**Review by tetromino — APPROVED** [review]

**Pull request merged**